test/e2e/traceflow_test.go

```diff
@@ -130,102 +322,73 @@ func TestTraceflow(t *testing.T) {
 		},
 	}
 
-	// Setup 2 NetworkPolicies:
-	// 1. Allow all egress traffic.
-	// 2. Deny ingress traffic on pod with label antrea-e2e = node1Pods[1]. So flow node1Pods[0] -> node1Pods[1] will be dropped.
-	var allowAllEgress *networkingv1.NetworkPolicy
-	allowAllEgressName := "test-networkpolicy-allow-all-egress"
-	if allowAllEgress, err = data.createNPAllowAllEgress(allowAllEgressName); err != nil {
-		t.Fatalf("Error when creating network policy: %v", err)
-	}
-	defer func() {
-		if err = data.deleteNetworkpolicy(allowAllEgress); err != nil {
-			t.Fatalf("Error when deleting network policy: %v", err)
-		}
-	}()
-
-	var denyAllIngress *networkingv1.NetworkPolicy
-	denyAllIngressName := "test-networkpolicy-deny-ingress"
-	if denyAllIngress, err = data.createNPDenyAllIngress("antrea-e2e", node1Pods[1], denyAllIngressName); err != nil {
-		t.Fatalf("Error when creating network policy: %v", err)
-	}
-	defer func() {
-		if err = data.deleteNetworkpolicy(denyAllIngress); err != nil {
-			t.Fatalf("Error when deleting network policy: %v", err)
-		}
-	}()
-
-	antreaPod, err := data.getAntreaPodOnNode(node1)
-	if err = data.waitForNetworkpolicyRealized(antreaPod, allowAllEgressName); err != nil {
-		t.Error(err)
-	}
-	if err = data.waitForNetworkpolicyRealized(antreaPod, denyAllIngressName); err != nil {
-		t.Error(err)
-	}
-
-	// Create 2 traceflows:
-	// 1. node1Pods[0] -> node1Pods[1], intra node1.
-	// 2. node1Pods[0] -> node2Pods[0], inter node1 and node2.
-	if intraNodeTraceflow, err = data.crdClient.OpsV1alpha1().Traceflows().Create(context.TODO(), intraNodeTraceflow, v1.CreateOptions{}); err != nil {
-		t.Fatalf("Error when creating traceflow: %v", err)
-	}
-	defer func() {
-		if err := data.crdClient.OpsV1alpha1().Traceflows().Delete(context.TODO(), intraNodeTraceflow.Name, v1.DeleteOptions{}); err != nil {
-			t.Fatalf("Error when deleting traceflow: %v", err)
-		}
-	}()
-
-	if interNodeTraceflow, err = data.crdClient.OpsV1alpha1().Traceflows().Create(context.TODO(), interNodeTraceflow, v1.CreateOptions{}); err != nil {
-		t.Fatalf("Error when creating traceflow: %v", err)
-	}
-	defer func() {
-		if err := data.crdClient.OpsV1alpha1().Traceflows().Delete(context.TODO(), interNodeTraceflow.Name, v1.DeleteOptions{}); err != nil {
-			t.Fatalf("Error when deleting traceflow: %v", err)
-		}
-	}()
+	t.Run("traceflowGroupTest", func(t *testing.T) {
+		for _, tc := range testcases {
+			tc := tc
+			t.Run(tc.name, func(t *testing.T) {
+				t.Parallel()
+				if _, err := data.crdClient.OpsV1alpha1().Traceflows().Create(context.TODO(), tc.tf, v1.CreateOptions{}); err != nil {
+					t.Fatalf("Error when creating traceflow: %v", err)
+				}
+				defer func() {
+					if err := data.crdClient.OpsV1alpha1().Traceflows().Delete(context.TODO(), tc.tf.Name, v1.DeleteOptions{}); err != nil {
+						t.Fatalf("Error when deleting traceflow: %v", err)
```

Comment:
nit: maybe `Error` instead of `Fatal` in the cleanup function

Reply:
Yes, thank you for pointing out this. I updated all `t.Fatal` and `t.Error` in the traceflow_test. Please take another look.